src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,195 @@
+    private static final int    PREFERRED_MAX_BARBARIANS            = 22;
+    private static final int    PREFERRED_MAX_ARCHERS               = 16;
+    private static final int    PREFERRED_MAX_CHIEFS                = 2;
+    private static final float  WHOLE_CIRCLE                       = 360.0F;
+    private static final float  HALF_A_CIRCLE                      = 180F;
+
+    /**
+     * Private constructor to hide the implicit public one.
+     */
+    private BarbarianUtils()
+    {
+    }
+
+    public static void doRaid(final World world, final int level, final Colony colony)
+    {
+        eventRaid(world, level, colony);
+    }
+
+    /**
+     * Commences a Raid event in a specific colony, with a specific RaidLevel
+     */
+
+    private static void eventRaid(final World raidingWorld, final int level, final Colony colony)
+    {
+        if (level == LEVEL_AT_WHICH_TO_NOT_TRIGGER_RAID)
+        {
+            return;
+        }
+
+        int numberOfBarbarians = (int) (BARBARIANS_MULTIPLIER * level) ;
+        int numberOfArcherBarbarians = (int) (ARCHER_BARBARIANS_MULTIPLIER * level);
+        int numberOfChiefBarbarians = (int) (CHIEF_BARBARIANS_MULTIPLIER * level);
+
+        int hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+        if (hordeTotal > PREFERRED_MAX_HORDE_SIZE && MAX_SIZE == PREFERRED_MAX_HORDE_SIZE && hordeTotal > MAX_SIZE)
+        {
+            //set the preferred horde style if the total spawns is greater that the config's max size
+            numberOfBarbarians = PREFERRED_MAX_BARBARIANS;
+            numberOfArcherBarbarians = PREFERRED_MAX_ARCHERS;
+            numberOfChiefBarbarians = PREFERRED_MAX_CHIEFS;
+        }
+        else if (hordeTotal > MAX_SIZE)
+        {
+            //Equilize the spawns so that there is less spawns than the config's max size
+            numberOfBarbarians = equilizeBarbarianSpawns(hordeTotal, numberOfBarbarians);
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+            numberOfArcherBarbarians = equilizeBarbarianSpawns(hordeTotal, numberOfArcherBarbarians);
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+            numberOfChiefBarbarians = equilizeBarbarianSpawns(hordeTotal, numberOfChiefBarbarians);
+        }
+
+        //Make sure world isn't null
+        if (raidingWorld == null)
+        {
+            return;
+        }
+
+        spawn(BarbarianUtils.barbarian, numberOfBarbarians, colony, raidingWorld);
+        spawn(BarbarianUtils.archer, numberOfArcherBarbarians, colony, raidingWorld);
+        spawn(BarbarianUtils.chief, numberOfChiefBarbarians, colony, raidingWorld);
+    }
+
+    private static int equilizeBarbarianSpawns(final int total, int numberOf)
+    {
+        if (total > Configurations.maxBarbarianHordeSize)
+        {
+
+            numberOf = total - MAX_SIZE;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "numberOf". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1226)